src/shadows.cpp

```diff
@@ -49,6 +48,10 @@ void Shadows::shadow_pass(Scene *scene) {
     glEnable(GL_CULL_FACE);
     glCullFace(GL_BACK);
     glBindFramebuffer(GL_FRAMEBUFFER, 0);
+
+    // Restore viewport.
+    glViewport(0, 0, screen_width, screen_height);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
```

Comment:
I think of this as the shadows module cleaning up after itself. Though it does seem a very windowy thing to do, certainly. Shall I just pass a reference to ``window`` down to ``render`` then?